src/Parsers/ASTFunction.cpp

```diff
@@ -37,94 +37,130 @@ namespace
 {
     /// Finds arguments of a specified function which should not be displayed for most users for security reasons.
     /// That involves passwords and secret keys.
-    /// The member function getRange() returns a pair of numbers [first, last) specifying arguments
-    /// which must be hidden. If the function returns {-1, -1} that means no arguments must be hidden.
     class FunctionSecretArgumentsFinder
     {
     public:
         explicit FunctionSecretArgumentsFinder(const ASTFunction & function_) : function(function_)
         {
-            if (function.arguments)
+            if (!function.arguments)
+                return;
+
+            const auto * expr_list = function.arguments->as<ASTExpressionList>();
+            if (!expr_list)
+                return;
+
+            arguments = &expr_list->children;
+            switch (function.kind)
             {
-                if (const auto * expr_list = function.arguments->as<ASTExpressionList>())
-                    arguments = &expr_list->children;
+                case ASTFunction::Kind::ORDINARY_FUNCTION: findOrdinaryFunctionSecretArguments(); break;
+                case ASTFunction::Kind::WINDOW_FUNCTION: break;
+                case ASTFunction::Kind::LAMBDA_FUNCTION:  break;
+                case ASTFunction::Kind::TABLE_ENGINE: findTableEngineSecretArguments(); break;
+                case ASTFunction::Kind::DATABASE_ENGINE: findDatabaseEngineSecretArguments(); break;
+                case ASTFunction::Kind::BACKUP_NAME: findBackupNameSecretArguments(); break;
             }
         }
 
-        std::pair<size_t, size_t> getRange() const
+        struct Result
         {
-            if (!arguments)
-                return npos;
+            /// Result constructed by default means no arguments will be hidden.
+            size_t start = static_cast<size_t>(-1);
+            size_t count = 0;
```

Comment:
For my understanding, what would be an example query where we have more than 1 field to be hidden ? 
Also I believe we assume that these field to be hidden will be consecutive arguments as we have a `start` & `count`, will it always be the case (excluding named collections) ?

Reply:
I see we use this for userid, password etc, and form the test examples looks like they are consecutive arguments.

Reply:
I see we have examples of `userid`, `password` etc and from the test examples looks like they are consecutive

Reply:
Hidden arguments are always consecutive. Though in general they don't have to be always consecutive, they're consecutive in all the known cases.

There only few cases where we have more than 1 argument to be hidden:

```
encrypt('mode', 'plaintext', 'key' [, iv, aad]) -> encrypt('mode', '[HIDDEN]')
remote('addresses_expr', db_or_db_table, a1, a2, a3) -> remote('addresses_expr, db_table, a1, 'HIDDEN', 'HIDDEN')
mysql(named_collection_name, password = 'password', user = 'user', password = 'password')
```

The last two cases need explanation: for `remote()` we hide both `a2` and `a3` if we cannot easily determine which of `a2` and `a3` is a password. And for `mysql` it doesn't actually make sense to specify the password multiple times, however it's allowed.

Reply:
I think I'll add more comments to the code.